flexx/app/_tornadoserver.py

```diff
@@ -678,13 +678,18 @@ def check_origin(self, origin):
         serving_host = self.request.headers.get("Host")
         serving_hostname = serving_host.split(':')[0]
         connecting_host = urlparse(origin).netloc
+        connecting_hostname = connecting_host.split(':')[0]
         
         if serving_hostname == 'localhost':
             return True  # Safe
         elif serving_hostname == '0.0.0.0':
             return True  # we cannot know if the origin matches
         elif serving_host == connecting_host:
             return True
+        elif serving_hostname == connecting_hostname:
+            # Comparing hostnames in case of Jupyter notebooks
+            # connecting_host and serving_host have same host names but different ports
+            return True
```

Comment:
> Taking one step back, why is the port number different? I have not experienced this before, can we do something to use the same port?

Typically Jupyter notebook is running on 8888 port, and flexx backend is on 49xxx port, when you are trying to connect to WebSocket from running notebook the ports are different, so this lead to CORS checks failure.